File: jetbrains-opnit/src/main/java/org/opnit/plugin/OpnitFileType.java
package org.opnit.plugin;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class OpnitFileType extends LanguageFileType {
    public static final OpnitFileType INSTANCE = new OpnitFileType();

    private OpnitFileType() {
        super(OpnitLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "Opnit File";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "Opnit language file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "opnit";
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return null;
    }
} 

File: intellj/src/main/kotlin/ru/openitstudio/language/OpnitFileType.kt
package ru.openitstudio.language

import com.intellij.openapi.fileTypes.LanguageFileType
import com.intellij.openapi.util.IconLoader
import javax.swing.Icon

class OpnitFileType private constructor() : LanguageFileType(OpnitLanguage.INSTANCE) {
    companion object {
        @JvmStatic
        val INSTANCE = OpnitFileType()
    }

    override fun getName(): String = "Opnit File"
    override fun getDescription(): String = "Opnit language file"
    override fun getDefaultExtension(): String = "opnit"
    override fun getIcon(): Icon = IconLoader.getIcon("/icons/icon.svg", OpnitFileType::class.java)
} 